From: Yaniv Rosner <yaniv.rosner@qlogic.com>
Date: Thu, 12 Jun 2014 07:55:30 +0300
Subject: [PATCH 219/242] bnx2x: Fix 1G-baseT link
Patch-mainline: v3.16-rc1

Git-commit: a2755be5b5340eeb0f00a318d23a302fe95c45ba
References: bsc#908684 FATE#317539


Set the phy access mode even in case of link-flap avoidance.

Signed-off-by: Yaniv Rosner <yaniv.rosner@qlogic.com>
Signed-off-by: Yuval Mintz <yuval.mintz@qlogic.com>
Signed-off-by: Ariel Elior <ariel.elior@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Cho, Yu-Chen <acho@suse.com>
---
 drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_link.c
@@ -12444,6 +12444,7 @@ static int bnx2x_avoid_link_flap(struct
 	u32 dont_clear_stat, lfa_sts;
 	struct bnx2x *bp = params->bp;
 
+	bnx2x_set_mdio_emac_per_phy(bp, params);
 	/* Sync the link parameters */
 	bnx2x_link_status_update(params, vars);
 
